src/serverExplorer.ts

```diff
@@ -115,12 +115,23 @@ export class ServersViewTreeDataProvider implements TreeDataProvider< Protocol.S
         view.reveal(data, { focus: true, select: true });
     }
 
-    public addDeployment(server: Protocol.ServerHandle): Thenable<Protocol.Status> {
+    public async addDeployment(server: Protocol.ServerHandle): Promise<Protocol.Status> {
+        // quickpick to solve a vscode api bug in windows that only opens file-picker dialog either in file or folder mode
+        const isWindows: boolean = process.platform.indexOf('win') >= 0;
+        let filePickerType = 'file or exploded';
```

Comment:
This methods gets rather lengthy (it already was but is now even more). I'd thus suggest to extract the windows specific part to it's own method (ex. #quickpickDeploymentType)

Reply:
I don't fully understand your suggestion. it doesn't seem to me we can extract much of the code to another method. Look at the last commit. This is the solution that maintains the code most readable possible for me.

Reply:
works perfectly for me, great way to separate things, too.